vit-servicing-station-lib/src/db/queries/search/mod.rs

```diff
@@ -0,0 +1,103 @@
+use diesel::{
+    expression_methods::ExpressionMethods,
+    r2d2::{ConnectionManager, PooledConnection},
+    QueryDsl, RunQueryDsl, TextExpressionMethods,
+};
+
+use crate::{
+    db::{DbConnection, DbConnectionPool},
+    v0::{
+        endpoints::search::requests::{Column, Constraint, OrderBy, Query, SearchResponse, Table},
+        errors::HandleError,
+    },
+};
+
+pub async fn search_db(
+    query: Query,
+    pool: &DbConnectionPool,
+) -> Result<SearchResponse, HandleError> {
+    let db_conn = pool.get().map_err(HandleError::DatabaseError)?;
+    tokio::task::spawn_blocking(move || search(query, &db_conn))
+        .await
+        .map_err(|_e| HandleError::InternalError("Error executing request".to_string()))?
+}
+
+fn search(
+    Query {
+        table,
+        filter,
+        order_by,
+    }: Query,
+    conn: &PooledConnection<ConnectionManager<DbConnection>>,
+) -> Result<SearchResponse, HandleError> {
+    match table {
+        Table::Challenges => {
+            use crate::db::schema::challenges::dsl::*;
+            use Column::*;
+
+            let mut query = challenges.into_boxed();
+
+            for Constraint { search, column } in filter {
+                let search = format!("%{search}%");
```

Comment:
mmm this is user input, how well are we guarded against malicious inputs?

Reply:
I googled it when I write it, and I found something saying diesel does input sanitization, I'll see if I can find that comment again

Reply:
https://github.com/diesel-rs/diesel/issues/229

Reply:
Seems like it does, it's just not super well documented